target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-deco-x80-5g.dts

```diff
@@ -0,0 +1,488 @@
+		pinctrl-names = "default";
+
+		reset_button {
+			label = "reset";
+			gpios = <&tlmm 27 GPIO_ACTIVE_LOW>; /*reset button output*/
+			linux,code = <KEY_RESTART>; /* Aliases for button to stitch to Linux reset function */
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-0 = <&led_pins>;
+		pinctrl-names = "default";
+		/*note the LED's defined here are not individual they are a single combined RGB element
+		to mimic the OEM default behaviour green and red are switched on from boot to indicate yellow
+		to show the device is booting behaviour can be user defined in procd to override this.
+		OEM behaviour of multi function LED:
+		pulse Yellow: resetting/ rebooting.
+		Solid Yellow: starting up.
+		Pulse Blue: factory reset/ first boot ready for setup or WPS in progress.
+		Solid Blue: setting up/ committing settings change.
+		Pulse Green: upgrading firmware.
+		Solid Green: connected to 4G / WAN.
+		Solid Purple: connected to 5G.
+		Pulse Red: mesh failure cant find peers.
+		Solid Red: fault */
+
+		led_system_green: green {
+			gpios = <&tlmm 50 GPIO_ACTIVE_HIGH>; /* green led component of rgb led */
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_STATUS;
+			default-state = "on"; /* default state on to select component of yellow led color on start-up */
+		};
+
+		led_system_red: red {
+			gpios = <&tlmm 51 GPIO_ACTIVE_HIGH>; /* red led component of rgb led */
+			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_STATUS;
+			default-state = "on";  /* default state on to select component of yellow led color on start-up */
+		};
+
+		led_system_blue: blue {
+			gpios = <&tlmm 52 GPIO_ACTIVE_HIGH>; /* blue led component of rgb */
+			color = <LED_COLOR_ID_BLUE>;
+			function = LED_FUNCTION_STATUS;
+		};
+	};
+
+	gpio_fan: gpio-fan {
+		compatible = "gpio-fan";
+		pinctrl-0 = <&fan_pins>;
+		pinctrl-names = "default";
+		gpios = <&tlmm 42 GPIO_ACTIVE_LOW>, /* fan gpio MSB */
+			<&tlmm 41 GPIO_ACTIVE_LOW>,
+			<&tlmm 40 GPIO_ACTIVE_LOW>; /* fan gpio LSB */
+
+		/* stock fan is a Sunnon EF40201BX-Q18C-F99
+		6800 rpm
+		gpio68 seemed to be the feedback rpm of the fan.
+		no support for fan speed in kernel saved for future use.*/
+
+		/*rpm-gpios = <&tlmm 68 GPIO_ACTIVE_HIGH>; */
+		gpio-fan,speed-map = <0 0>, /* speed 0 binary drive bits 0,0,0 (idle state) */
+				<971 1>, /* speed 1 binary drive bits 0,0,1      ▼ */
+				<1942 2>, /* speed 3 binary drive bits 0,1,0 (active state) */
+				<2913 3>, /* speed 4 binary drive bits 0,1,1      ▼ */
+				<3884 4>, /* speed 5 binary drive bits 1,0,0      ▼ */
+				<4855 5>, /* speed 6 binary drive bits 1,0,1 (high state) */
+				<5826 6>, /* speed 7 binary drive bits 1,1,0      ▼ */
+				<6800 7>; /* speed 8 binary drive bits 1,1,1      ▼ */
+		/* alarm-gpios = <&tlmm XX GPIO_ACTIVE_HIGH>; */
+		cooling-min-level = <0>; /* minimum speed allowable to be set via thermal */
+		cooling-max-level = <7>; /* max speed allowable to be set via thermal */
```

Comment:
I dont think `cooling-min-level` and `cooling-max-level` exist upstream, bindings dont describe them at all